From e4d483a2ef99a71c6fcd6ad2e6c2f8f88ba380f4 Mon Sep 17 00:00:00 2001
From: Vaibhav Devmurari <vdevmurari@google.com>
Date: Mon, 9 Dec 2024 13:15:03 +0000
Subject: [PATCH] Fix: Block opening settings app on keyguard without user auth

Test: atest KeyGestureEventTests
Bug: 378900798
Flag: EXEMPT bugfix
Change-Id: I89d43872108710d20e0c4ef7e652d389896155d7
(cherry picked from commit d615298466085c4a88c6733804160e0c1ee7e31e)
---
 .../core/java/com/android/server/policy/PhoneWindowManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index bf90a6c1926c..d24a4d914c83 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -3447,7 +3447,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 }
                 break;
             case KeyEvent.KEYCODE_I:
-                if (firstDown && event.isMetaPressed()) {
+                if (firstDown && event.isMetaPressed() && isUserSetupComplete() && !keyguardOn) {
                     showSystemSettings();
                     logKeyboardSystemsEvent(event, KeyboardLogEvent.LAUNCH_SYSTEM_SETTINGS);
                     return true;
-- 
2.50.1.351.gcf940e82a1-goog

